file_windows.go

```diff
@@ -0,0 +1,33 @@
+package write
+
+import (
+	"os"
+	"syscall"
+)
+
+const (
+	movefile_replace_existing = 0x1
+	movefile_write_through    = 0x8
+)
+
+//sys moveFileEx(lpExistingFileName *uint16, lpNewFileName *uint16, dwFlags uint32) (err error) = MoveFileExW
+
+// replaceFile atomically replaces the destination file or directory with the
+// source.  It is guaranteed to either replace the target file entirely, or not
+// change either file.
+func replaceFile(source, destination string) error {
+	src, err := syscall.UTF16PtrFromString(source)
+	if err != nil {
+		return &os.LinkError{"replace", source, destination, err}
```

Comment:
here and below: could you please change the literal construction to use field names to be robust when os.LinkError changes? E.g. `&os.LinkError{Op: "replace", Old: source, New: destination, Err: err}`

Reply:
Ahh yeah, definitely.

Reply:
Let me know once you pushed the change :)